packages/bpk-component-graphic-promo/src/BpkGraphicPromo.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkText from 'bpk-component-text';
+import BpkButton from 'bpk-component-button';
+import BpkCard from 'bpk-component-card';
+
+import TEXT_ALIGN from './BpkGraphicPromoTextAlign';
+import TEXT_COLORS from './BpkGraphicPromoTextColors';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  className: ?string,
```

Comment:
Why is this a Maybe rather than an optional? 

`className: ?string`
https://flow.org/en/docs/types/maybe/

vs

`className?: string`
https://flow.org/en/docs/types/objects/

Reply:
+1 `className` should always be optional to a component

Reply:
It looks like convention in Backpack to provide `null` as the default value for optional `props` that do not have a default value otherwise (see `PropTypes` at the bottom of the file). My personal opinion is to use `undefined` for primitives or missing arguments and `null` for objects, but I tried to stay consistent with the existing codebase. If I make this optional, the type would not be accurate unfortunately, because it can be `null` | `undefined` if not set.

What do you think?

Reply:
For optional props, inside of non-Backpack, we'd either provide a sensible default, or assert of the falsey nature of `undefined` - not use `null` as a default value to then assert on for being falsey. `null` is used in places where we want to explicitly set something as missing/falsey, but by a consumer, not as a default. Will link to internal thread/docs.  

That however may not be the standard in this repo?

Reply:
Ideally, we could banish `null` from the codebase and only use `undefined`. That will reduce our 2B-dollar mistake into a 1B-dollar mistake (quoting Tony Hoare 😂). However, I am happy to do whatever you prefer in this codebase. 🙂

Reply:
@dominicfraser Any ideas on this one? It seems that this repository uses `null` as the default value. 😅